### PR TITLE
Editor ui issues

### PR DESCRIPTION
**Pull request by EvidentlyCube**
- It is no longer possible to change map's display type for a hold you don't own
- It is no longer possible to change map image for a hold you don't own
- When in EditSelectScreen you change from a hold that has world map to one that is not editable, map buttons become disabled now
- It's no longer possible to change 'Show Exit level' for a hold you don't own
- UI elements in EditSelectScreen will no longer be redrawn on top of different elements when changing tabs and focus is on the element outside the tabbed interface

[Relevant thread](http://forum.caravelgames.com/viewtopic.php?TopicID=42247)

### CODE REVIEW

File: FrontEndLib/Widget.h [hunk 1]
```diff
@@ -372,7 +372,7 @@ class CWidget : public CAttachableObject
 	UINT        GetW() const {return this->w;}
 	UINT        GetH() const {return this->h;}
 
-	virtual void   Hide(const bool /*bPaint*/ = true) {this->bIsVisible = false;}
+	virtual void   Hide(const bool /*bPaint*/ = true);
```
Comment (mrimer) [inline FrontEndLib/Widget.h:375]: Remove commenting around bPaint here.

File: FrontEndLib/Widget.cpp [hunk 2]
```diff
@@ -684,6 +684,20 @@ const
 	ChildContainerRect.y += this->nChildrenScrollOffsetY;
 }
 
+//***************************************************************************
+void CWidget::Show(const bool bFlag)
+{
+	this->bIsVisible = bFlag;
+	if (!bFlag)
+		this->Hide(false);
```
Comment (mrimer) [inline FrontEndLib/Widget.cpp:692]: remove `this->` ?

**Review by mrimer — APPROVED** [review]
Thank you!  Looks good.

Couple stylistic suggestions.

File: FrontEndLib/Widget.h (hunk 1 above)
Comment (EvidentlyCube) [reply FrontEndLib/Widget.h:375]: Why was that commenting there in the first place? I've always been curious whenever I've encountered it. Was it just to stress that the default implementation does not use the argument?

**Comment by EvidentlyCube** [comment]
PR updated

**Review by mrimer — APPROVED** [review]

File: FrontEndLib/Widget.h (hunk 1 above)
Comment (mrimer) [reply FrontEndLib/Widget.h]: > Why was that commenting there in the first place? I've always been curious whenever I've encountered it. Was it just to stress that the default implementation does not use the argument?

@EvidentlyCube , great question.  This is actually to suppress a compiler warning.  Any time a method (virtual, in this case) has named arguments that aren't referenced in the body, the compiler may complain about an unused variable.  The way to suppress the variable is to not name the variable in the function signature that's alongside the definition.

**Pull request merged**